### PR TITLE
feat: flag to set dataflow template

### PR DESCRIPTION
**Pull request by asthamohta**

### CODE REVIEW

**Review by manitgupta — COMMENTED** [review]
Add it to the web command as well

File: webv2/webCmd.go [hunk 1]
```diff
@@ -56,10 +58,12 @@ func (cmd *WebCmd) SetFlags(f *flag.FlagSet) {
 	f.BoolVar(&cmd.open, "open", false, "Opens the Spanner migration tool web interface in the default browser, defaults to false")
 	f.IntVar(&cmd.port, "port", 8080, "The port in which Spanner migration tool will run, defaults to 8080")
 	f.BoolVar(&cmd.validate, "validate", false, "Flag for validating if all the required input parameters are present")
+	f.StringVar(&cmd.dataflowTemplate, "dataflow-template", constants.DEFAULT_TEMPLATE_PATH, "GCS path of the Dataflow template")
```
Comment (manitgupta) [inline webv2/webCmd.go:61]: Add a UT?

File: cmd/data.go [hunk 2]
```diff
@@ -89,6 +90,7 @@ func (cmd *DataCmd) SetFlags(f *flag.FlagSet) {
 	f.StringVar(&cmd.logLevel, "log-level", "DEBUG", "Configure the logging level for the command (INFO, DEBUG), defaults to DEBUG")
 	f.BoolVar(&cmd.SkipForeignKeys, "skip-foreign-keys", false, "Skip creating foreign keys after data migration is complete (ddl statements for foreign keys can still be found in the downloaded schema.ddl.txt file and the same can be applied separately)")
 	f.BoolVar(&cmd.validate, "validate", false, "Flag for validating if all the required input parameters are present")
+	f.StringVar(&cmd.dataflowTemplate, "dataflow-template", constants.DEFAULT_TEMPLATE_PATH, "GCS path of the Dataflow template")
```
Comment (aksharauke) [inline cmd/data.go:93]: Why is the default set to "dataflow-template"? Can it be set as default to DEFAULT_TEMPLATE_PATH from the Constants file? Same comment for other places where this flag is declared.

File: webv2/webCmd.go [hunk 3]
```diff
@@ -24,17 +24,19 @@ import (
...
-	logLevel string
-	open     bool
-	port     int
-	validate bool
+	DistDir  			embed.FS
+	logLevel 			string
+	open     			bool
+	port     			int
+	validate 			bool
+	dataflowTemplate 	string
```
Comment (aksharauke) [inline webv2/webCmd.go:39]: The docs section should also be kept in-sync with the new options added.

File: common/constants/constants.go [hunk 4]
```diff
@@ -101,4 +101,6 @@ const (
 	// Metadata table names
 	SMT_JOB_TABLE      string = "SMT_JOB"
 	SMT_RESOURCE_TABLE string = "SMT_RESOURCE"
+	// Default gcs path of the Dataflow template.
+	DEFAULT_TEMPLATE_PATH string = "gs://dataflow-templates-southamerica-west1/2024-03-22-02_RC00/flex/Cloud_Datastream_to_Spanner"
```
Comment (aksharauke) [inline common/constants/constants.go:105]: Please refer the latest comments on https://b.corp.google.com/issues/331514027. As per that, it should always be latest.

File: cmd/data.go (hunk 2 above)
Comment (asthamohta) [reply cmd/data.go:93]: Its not set to dataflow-template, that's the name of the flag

**Review by manitgupta — APPROVED** [review]

**Pull request merged**